### PR TITLE
ReadPDFFileV2 unescaping

### PR DESCRIPTION
**Pull request by anas-yousef**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/XSUP-32895

## Description
Added handling of html encoded URLs, by using **html.unescape**

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CommonScripts/ReleaseNotes/1_14_20.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "breakingChanges": true,
+    "breakingChangesNotes": "The script ReadPDFFileV2 now uses the argument *unescape*, with a default value of *True*. This update will change the default behavior of the script. System playbooks such as: *Extract Indicators From File - Generic v2*, and custom playbooks that use this script will be affected."
```
Comment (JasBeilin) [inline Packs/CommonScripts/ReleaseNotes/1_14_20.json:3]:
```suggestion
    "breakingChangesNotes": "The ***ReadPDFFileV2*** script now unescapes special character when extracting URLs by default. This update will affect the behavior of the script and of playbooks using it such as *Extract Indicators From File - Generic v2*, or any custom playbooks using it. In order to disable this behavior, set the *unescape* argument to `false`."
```

File: Packs/CommonScripts/ReleaseNotes/1_14_20.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+#### Scripts
+
+##### ReadPDFFileV2
+- Updated the Docker image to: *demisto/readpdf:1.0.0.89247*.
+
+- Added the argument **unescape**. When set to *true*, the script will unescape the URLs that have been escaped (Ignoring them). Default is 'True'.
```
Comment (JasBeilin) [inline Packs/CommonScripts/ReleaseNotes/1_14_20.md:7]:
```suggestion
- Added the argument **unescape**. When set, the script will unescape the URLs that have been escaped (by ignoring them). Default is 'true'. 
```

File: Packs/CommonScripts/Scripts/ReadPDFFileV2/ReadPDFFileV2.py [hunk 3]
```diff
@@ -239,9 +240,12 @@ def get_pdf_htmls_content(pdf_path: str, output_folder: str) -> str:
...
     html_content = ""
     for file_name in html_file_names:
-        with open(file_name, "rb") as f:
+        with open(file_name) as f:
             for line in f:
-                html_content += str(line)
+                if unescape:
+                    html_content += html.unescape(line)
+                else:
+                    html_content += line
```
Comment (JasBeilin) [inline Packs/CommonScripts/Scripts/ReadPDFFileV2/ReadPDFFileV2.py:248]:
```suggestion
                html_content += html.unescape(line) if unescape else line
```

File: Packs/CommonScripts/Scripts/ReadPDFFileV2/ReadPDFFileV2.py [hunk 4]
```diff
@@ -614,7 +620,8 @@ def handling_pdf_credentials(cpy_file_path: str, dec_file_path: str, encrypted:
     return cpy_file_path
 
 
-def extract_data_from_pdf(path: str, user_password: str, entry_id: str, max_images: int | None, working_dir: str) -> None:
+def extract_data_from_pdf(path: str, user_password: str, entry_id: str, max_images: int | None, working_dir: str,
+                          unescape: bool = True) -> None:
```
Comment (JasBeilin) [inline Packs/CommonScripts/Scripts/ReadPDFFileV2/ReadPDFFileV2.py:624]: I think I would change the parameter name to unescape_url, WDYT?

File: Packs/CommonScripts/Scripts/ReadPDFFileV2/ReadPDFFileV2.yml [hunk 5]
```diff
@@ -8,6 +8,13 @@ args:
 - defaultValue: "20"
   description: Maximum number of images to extract from the PDF file.
   name: maxImages
+- auto: PREDEFINED
+  defaultValue: 'true'
+  description: To unescape URLs that have been escaped as part of the URLs extraction. Invalid characters will be ignored.
+  name: unescape
```
Comment (JasBeilin) [inline Packs/CommonScripts/Scripts/ReadPDFFileV2/ReadPDFFileV2.yml:14]: I think I would change the parameter name to unescape_url, WDYT?

**Review by JasBeilin — CHANGES_REQUESTED** [review]
I got some minor corrections but overall looks good.

**Review by yuvalbenshalom — APPROVED** [review]

**Review by JasBeilin — APPROVED** [review]
Nice

**Pull request merged**